src/main/java/com/google/cloud/genomics/dataflow/readers/bam/Reader.java

```diff
@@ -75,11 +91,11 @@ public void process() throws IOException {
 
   void openFile() throws IOException {
     LOG.info("Processing shard " + shard);
-    final SamReader reader = BAMIO.openBAM(storageClient, shard. file, stringency);
+    final SamReader reader = BAMIO.openBAM(storageClient, shard. file, 
+        options.getStringency());
     iterator = null;
-    procesingUnmapped = shard.contig.referenceName.equals("*");
     if (reader.hasIndex() && reader.indexing() != null) {
-      if (procesingUnmapped) {
+      if (filter == Filter.UNMAPPED_ONLY) {
```

Comment:
You guys should take a look at this open issue in htsjdk: https://github.com/samtools/htsjdk/issues/278 (have not thought about the implications for this branch)

Basically, `queryUnmapped()` only returns a subset of the total reads that are marked as unmapped. Unmapped reads with mapped mates are assigned the positions of their mates and so are interleaved with the mapped reads.